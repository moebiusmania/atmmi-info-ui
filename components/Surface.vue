<script lang="ts" setup>
const resource: string = "/api/surface";

const { data, error } = await useFetch<Array<string>>(resource);

if (error.value) console.error("ERROR from useFetch: ", error.value);
</script>

<template>
  <Card title="Linee di superficie">
    <p v-if="data && data.length > 0" v-for="item in data">{{ item }}</p>
    <p v-else>☀️ Nessun avviso disponibile</p>
  </Card>
</template>
