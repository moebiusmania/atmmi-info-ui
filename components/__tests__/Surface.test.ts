import { mount } from "@vue/test-utils";
import { expect, test, vi, describe } from "vitest";
import Surface from "../Surface.vue";

// Mock the Card component
vi.mock("../Card.vue", () => ({
  default: {
    name: "Card",
    template: '<div class="card-mock"><slot /></div>',
    props: ["title"],
  },
}));

// Mock useFetch
const mockUseFetch = vi.fn();
vi.mock("#imports", () => ({
  useFetch: () => mockUseFetch(),
}));

describe("<Surface />", () => {
  test("renders with data", () => {
    mockUseFetch.mockReturnValue({
      data: { value: ["Bus 42: deviazione", "Tram 14: interruzione"] },
      error: { value: null },
    });

    const wrapper = mount(Surface);
    expect(wrapper.html()).toMatchSnapshot("with-data");
  });

  test("renders empty state", () => {
    mockUseFetch.mockReturnValue({
      data: { value: [] },
      error: { value: null },
    });

    const wrapper = mount(Surface);
    expect(wrapper.html()).toMatchSnapshot("empty-state");
  });
});
