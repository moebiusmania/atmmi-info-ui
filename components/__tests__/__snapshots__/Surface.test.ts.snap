// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Surface /> > renders empty state > empty-state 1`] = `""`;

exports[`<Surface /> > renders with data > with-data 1`] = `""`;
